src/components/Modal/VoteProposalModal.tsx

```diff
@@ -30,134 +51,139 @@ export const VoteProposalModal: FC<Props> = ({
   errorMessage,
 }) => {
   const [vote, setVoting] = useState<Vote | null>(null)
+  const [copied, setCopied] = useState(false)
+
   const handleSubmit = (e: any) => {
     e.preventDefault()
     if (vote) {
       onSubmit(vote)
     }
   }
+
+  const handleCopy = () => {
+    navigator.clipboard.writeText(address)
+    setCopied(true)
+    setTimeout(() => setCopied(false), 2000) // Reset after 2 seconds
```

Comment:
I would go with 3s instead, what do you think?

Reply:
I don't have a strong opinion about the timing, so 3s works too.